std/arith.asm

```diff
@@ -140,25 +302,25 @@ machine Arith(CLK32_31, operation_id){
     *
     *******/
 
-    // 0xffffffffffffffffffffffffffffffffffffffffffffffffffff fffe ffff fc2f
-    let p = array_as_fun([
-        0xfc2f, 0xffff, 0xfffe, 0xffff, 0xffff, 0xffff, 0xffff, 0xffff,
-        0xffff, 0xffff, 0xffff, 0xffff, 0xffff, 0xffff, 0xffff, 0xffff
-    ]);
+    let p: int -> expr = |i| if i >= 0 && i < 16 {
+        expr((secp_modulus >> (i * 16)) & 0xffff)
```

Comment:
We could use a byte_slice function here (and also use that for select_limb?)

Reply:
You mean a function that can select arbitrary bit ranges? Seems a bit overkill, no?

I just tried having a `select_limb: int, int -> int` function and using it here. I'm getting a stack overflow! I'll look into why, hopefully it's something we can fix.

Reply:
I just tried the same thing again and it worked 😮 Maybe I did something wrong the other day.

Reply:
The main problem I have with this is matching `i * 16` with `& 0xffff`, but it's fine.